src/java.base/share/classes/java/lang/invoke/LambdaForm.java

```diff
@@ -1068,20 +1095,28 @@ boolean contains(Name name) {
         private @Stable MethodHandle resolvedHandle;
         @Stable MethodHandle invoker;
         private final MethodHandleImpl.Intrinsic intrinsicName;
+        private final Object intrinsicData;
```

Comment:
This seems like the wrong place to add arbitrary data related only to intrinsics. Could this be moved either into `MethodHandleImpl$IntrinsicMethodHandle` or - perhaps less appealingly - rewrite the `MethodHandleImpl.Intrinsic` to a regular class which can then be instantiated with extra data? That `NamedFunction` holds a field of type `MethodHandleImpl.Intrinsic` instead of delegating to `resolvedHandle.intrinsicName()` seem like a pre-existing kludge that would be good to sort out why/if it's really necessary.